core/src/main/java/io/kestra/core/runners/RunContext.java

```diff
@@ -426,6 +426,26 @@ public RunContext forWorker(ApplicationContext applicationContext, WorkerTrigger
         return forScheduler(workerTrigger.getTriggerContext(), workerTrigger.getTrigger());
     }
 
+    public RunContext forWorkerDirectory(ApplicationContext applicationContext, WorkerTask workerTask) {
+        return forWorker(applicationContext, workerTask).withWorkerDirectoryTaskrun();
+    }
+
+    public RunContext withWorkerDirectoryTaskrun() {
+        Map<String, Object> clone = new HashMap<>(this.variables);
+
+        clone.put("workerTaskrun", clone.get("taskrun"));
+        if (clone.containsKey("parents")) {
+            clone.put("parents", clone.get("parents"));
+        }
+        if (clone.containsKey("parent")) {
+            clone.put("parent", clone.get("parent"));
+        }
+
+        this.variables = ImmutableMap.copyOf(clone);
+
+        return this;
+    }
```

Comment:
Can you merge those two methods in one ?

We really don't want a method that allow mutation of variables to be public!